docs/en/jpql-with-kotlin-jdsl/paths.md

```diff
@@ -13,6 +13,75 @@ entity(Book::class, "b").path(Book::isbn).path(Isbn::value)
 entity(Book::class, "b")(Book::isbn)(Isbn::value)
 ```
 
+## Java entity
+
+`path()` and `invoke()` take `KProperty1` or `KFuction1` as an argument.
+`KFunction1` is useful when you use Java entity with private property and public getter.
+
+```java
+@Entity
+public class Book {
+    @Id
+    private Long id;
+
+    private String title;
+
+    public String getTitle() {
+        return title;
+    }
+}
+```
+
+```kotlin
+// compile error
+path(Book::title)
+
+// Book.title
+path(Book::getTitle)
+```
+
+Kotlin JDSL follows the following rules to infer property name from getter name.
+
+- If the name starts with `is`, use the name as it is.
+- If the name starts with `get`, remove `get` and change the first letter to lowercase.
+- Otherwise, use the name as it is.
+
+```kotlin
+// Book.isAvailable
+path(Book::isAvailable)
+
+// Book.available
+path(Book::getAvailable)
+```
+
+If you want to use your own rule instead of the above rules, you need to implement `JpqlPropertyIntrospector` and provide it to `RenderContext`.
```

Comment:
I would like to change `you need to implement` to `you can implement`.

Reply:
It would be nice to leave a guide and link to the Custom DSL for more details, and to Spring Support if the user is using Spring.

Reply:
Also it would be nice to add Spring AutoConfigure, since there is no Spring Support for `JpqlIntrospector`. This forces the user to modify the `JpqlRenderContext` by themselves, which I think will be inconvenient for the user.

```kotlin
open class KotlinJdslAutoConfiguration {
    @Bean
    @ConditionalOnMissingBean
    @SinceJdsl("3.0.0")
    open fun jpqlRenderContext(
        serializers: List<JpqlSerializer<*>>,
        introspectors: List<JpqlIntrospector>,
    ): JpqlRenderContext {
        val userDefinedSerializers = object : JpqlRenderModule {
            override fun setupModule(context: JpqlRenderModule.SetupContext) {
                context.addAllSerializer(serializers.reversed())

                introspectors.reversed().forEach {
                    context.prependIntrospector(it)
                }
            }
        }

        return JpqlRenderContext()
            .registerModules(userDefinedSerializers)
    }
    // ...
}
```